csrc/scheduler/matmul.cpp

```diff
@@ -1021,16 +1001,14 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
   // Propagate tiling globally
   scheduler_utils::transformPropagateToAllFrom(mma_result, -1);
 
-  if (params.use_smem_epilogue) {
-    // TODO: this is no longer needed because we are now using the new swizzle
-    // for the smem epilogue. However, without this, I am seeing some failures
-    // in IdModel. Leaving this as is for now. Will investigate later.
-
-    // Transform mma_result through the epilogue swizzle without actually
-    // swizzling the axes. This is done to enable the domains
-    // are mapped between mma_result and smem_epilogue.
-    auto swizzled_dom = swizzleSharedMemory(mma_result);
-    mma_result->setLoopDomain(swizzled_dom.as<IterDomain*>());
+  if (params.use_smem_epilogue && params.splitk_factor != 1) {
+    // TODO:
+    // This is a workaround for a problem that different dimensions in the loop
+    // domain are mapped in the loop graph of IdModel due to the mapping of
+    // compliment IDs. We should remove forwarding completely, and remove this
+    // workaround.
+    mma_result->split(-2, 1);
+    mma_result->merge(-3);
   }
```

Comment:
If I understand correctly, this is splitting the second to last dimension (`Ni`) and merging it back, so that it does not get loop mapped with `smem_epilogue`?

Reply:
Yes, this is splitting the `Ni` and merging it back. The purpose is not to prevent it from being mapped to `smem_epilogue`.  But to prevent different dimensions of the loop domain of the bias tensor being mapped together. Generally, when we have a loop domain of a tensor as `[I1, I2, I3]`, then `I1`, `I2` and `I3` should never be mapped with each other in the loop graph, because by definition, they are different loops. However, due to the mapping of compliment IDs (which is a workaround introduced to fix loop promotion by mapping non-equivalent IDs together), these  IDs might be mapped together, which is clearly a mistake. I did not study in more detail why mapping compliment IDs could cause trouble in the bias tensor, but whatever reason it is, I believe forwarding and mapping compliment IDs is just fundamental wrong, and any effort that tries to "improve" it is a waste of time. So I decided to save my time and pick whatever workaround that takes me least amount of time to implement.